sentinel-adapter/sentinel-apache-dubbo-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/dubbo/BaseSentinelDubboFilter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.csp.sentinel.adapter.dubbo;
+
+
+import com.alibaba.csp.sentinel.AsyncEntry;
+import com.alibaba.csp.sentinel.Entry;
+import com.alibaba.csp.sentinel.Tracer;
+import com.alibaba.csp.sentinel.context.ContextUtil;
+import org.apache.dubbo.rpc.Filter;
+import org.apache.dubbo.rpc.Invocation;
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.Result;
+import org.apache.dubbo.rpc.RpcContext;
+
+/**
+ * Bas:e Class of the {@link SentinelDubboProviderFilter} and {@link SentinelDubboConsumerFilter}.
+ *
+ * @author Zechao Zheng
+ */
+public abstract class BaseSentinelDubboFilter implements Filter {
+
+    @Override
+    public Result onResponse(Result appResponse, Invoker<?> invoker, Invocation invocation) {
+        //it's unnecessary to traceAndExit the business exception
+        traceAndExit(null, invocation);
```

Comment:
Why it's not necessary to trace the biz exception? I've tested three different scenarios:

- Normal responses will be handled in the `Filter#onResponse` callback (`whenComplete` callback will not catch it)
- Timeout. The `org.apache.dubbo.remoting.TimeoutException` will be caught in the `whenComplete` callback (`Filter#onResponse` will not catch it)
- Biz exception. Both `Filter#onResponse` and the `whenComplete` callback could get signaled, but the `Filter#onResponse` will be notified first, so if we don't trace the biz exception here, it will be lost. We might need to trace the `appResponse.getException()` here.